vitrina/projects/views.py

```diff
@@ -93,4 +104,71 @@ class ProjectHistoryView(HistoryView):
     model = Project
     detail_url_name = 'project-detail'
     history_url_name = 'project-history'
-    tabs_template_name = 'component/tabs.html'
+    tabs_template_name = 'vitrina/projects/tabs.html'
+
+
+class ProjectDatasetsView(HistoryMixin, ListView):
+    model = Dataset
+    template_name = 'vitrina/projects/datasets.html'
+    paginate_by = 20
+
+    object: Dataset
+    detail_url_name = 'project-detail'
+    history_url_name = 'project-history'
+
+    def dispatch(self, request, *args, **kwargs):
+        self.object = get_object_or_404(Project, pk=kwargs['pk'])
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_queryset(self):
+        return self.object.datasets.all()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['project'] = self.object
+        context['can_update_project'] = has_perm(
+            self.request.user,
+            Action.UPDATE,
+            self.object
+        )
+        return context
+
+
+class ProjectAddDatasetView(
+    LoginRequiredMixin,
+    PermissionRequiredMixin,
+    RevisionMixin,
+    UpdateView
+):
+    model = Project
+    form_class = AddDatasetForm
+    template_name = 'base_form.html'
+
+    def has_permission(self):
+        project = self.get_object()
+        return has_perm(self.request.user, Action.UPDATE, project)
+
+    def form_valid(self, form):
+        super().form_valid(form)
+        self.object = form.save(commit=False)
+        print(self.object)
+        for dataset in form.cleaned_data['dataset']:
+            self.object.datasets.add(dataset)
+        self.object.save()
+        return HttpResponseRedirect(reverse('project-datasets', kwargs={'pk': self.object.pk}))
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        context_data['current_title'] = _('Duomenų rinkinių pridėjimas')
+        return context_data
+
+
+class RemoveDatasetView(LoginRequiredMixin, PermissionRequiredMixin, View):
```

Comment:
Maybe use `DeleteView` with `confirm_delete.html` template as in `DeleteMemberView` for better consistency?